src/partisan_hiscamp_membership_strategy.erl

```diff
@@ -0,0 +1,253 @@
+        %% tracing is a macro
+        case partisan_config:get(tracing, ?TRACING) of
+            true ->
+                lager:info("~p: Forwarding subscription for ~p to node: ~p", [node(), Node, N]);
+            false ->
+                ok
+            end,
+            OM ++ [{N, {protocol, {forward_subscription, Node}}}]
+            end, OutgoingMessages1, Membership0),
+    C = partisan_config:get(scamp_c, ?SCAMP_C_VALUE),
+    ForwardMessages = lists:map(fun(N) ->
+        case partisan_config:get(tracing, ?TRACING) of
+            true ->
+                lager:info("~p: Forwarding additional subs. for ~p to node: ~p", [node(), Node, N]);
+            false ->
+                ok
+        end,
+        {N, {protocol, {forward_subscription, Node}}}
+        end, select_random_sublist(state0, C)),
+    OutgoingMessages = OutgoingMessages2 ++ ForwardMessages,
+    State = State0#hiScamp{membership=Membership},
+    MembershipList = membership_list(State),
+    {ok, Membership, OutgoingMessages, State}.
+
+try_distance(#hiScamp{membership=Membership0, distance=Dist}=_State0, Node) ->
+    Dist = 0,
+    Dict = get(distance_metrics),
+    try dict:find(Node, Dict) of
+        {ok, Dist} -> Dist
+    catch
+        {error, Error} -> Error
+    end.
+    
+
+distance(#hiScamp{membership=Membership0, heap=Heap, level1=L1, level2=L2, distance = Dist}=_State0, Node) ->
+    % A node j joins the system by sending a subscription request to the node s which is closest to it
+    Threshold = 3,
+    try try_distance() of
+        Dist -> Dist
+    catch
+        error: Error ->
+            Error
+    end,
+    case Threshold > Dist of
+        true -> 
+            % process this case using Scamp within this cluster
+            % subscription at level 1
+            %need to send copies out to other nodes -> gossip protocol
+            sets:add(Node, L1);
+        false ->
+                       % search through set to find another set
+                       % store two sets in the record
+            % subscription at level 2
+                       %input Cluster0
+            sets:add(Node, L2)
+    end,
+    heaps:add(Dist, Heap).
+
+get_centroid_two_clusters(#hiScamp{membership=Membership0, heap=Heap, level1=L1, level2=L2} =State0) ->
+    Min1 = heaps:min(Heap),
+    Heap = heaps:delete_min(Heap),
+    Min2 = heaps:min(Heap),
+    case (sets:is_element(Min1, L1) andalso sets:is_element(Min2, L1) orelse 
+         (sets:is_element(Min1, L2) andalso sets:is_element(Min2, L2))) of
+        true ->
+            % can group nodes as theyre the closest within the same threshold
+        %% NODE CONNECTIONS
+            ok;
+        false -> 
+            Heap = heaps:merge(heaps:from_list(sets:to_list(L2), sets:to_list(L1)))
+    end.
+
+hierarchial_clustering(#hiScamp{membership=Memberhsip0, heap=Heap, level1=L1, level2=L2}=State0) ->
+    ok.
+
+leave(#hiScamp{membership=Membership0}=State0, Node) ->
```

Comment:
When a node leaves, you're not changing the levels at all, but you've altered the membership.  I think this does not match the protocol specification.